### ISSUE TITLE
Interactive selection of scale domain reorders (sorts) the data

### ISSUE THREAD

**Issue by nadluru** [vega/vega-lite#4221]
Dear Vega-lite team,

I am trying to visualize an overview-detail type of plot. When I do an interval selection on the nominal data, the data in the selected domain seems to be sorted even though I specify sort as null for the encoding channel. And also several of the data points get mapped to "undefined" which I can live with but is there a way to keep the data unsorted after selecting an interval?

Thanks so much for your help with this!

The vega-lite scheme (which should work on the vega-lite editor) is pasted below:

```
 {
    "$schema": "https://vega.github.io/schema/vega-lite/v3.0.0-rc6.json",
    "data": {"values":[
        {
            "ID": "1234", 
            "Age": "79.03", 
            "Gender": "Male", 
            "DateTime": "1/10/10 9:07"
        }, 
        {
            "ID": "5678", 
            "Age": "59.34", 
            "Gender": "Female", 
            "DateTime": "8/15/10 10:15"
        }, 
        {
            "ID": "8901", 
            "Age": "46.2", 
            "Gender": "Female", 
            "DateTime": "7/19/10 6:05"
        }, 
        {
            "ID": "0246", 
            "Age": "62.79", 
            "Gender": "Female", 
            "DateTime": "1/18/10 4:12"
        }, 
        {
            "ID": "0246", 
            "Age": "64.68", 
            "Gender": "Female", 
            "DateTime": "12/23/10 2:34"
        }, 
        {
            "ID": "5678", 
            "Age": "71.56", 
            "Gender": "Female", 
            "DateTime": "6/17/12 9:46"
        }, 
        {
            "ID": "7890", 
            "Age": "65.55", 
            "Gender": "Male", 
            "DateTime": "12/13/12 2:12"
        }, 
        {
            "ID": "7890", 
            "Age": "65.96", 
            "Gender": "Male", 
            "DateTime": "7/13/11 2:43"
        }, 
        {
            "ID": "7890", 
            "Age": "65.05", 
            "Gender": "Male", 
            "DateTime": "8/13/14 6:02"
        }, 
        {
            "ID": "7890", 
            "Age": "34.05", 
            "Gender": "Male", 
            "DateTime": "6/13/13 6:02"
        }, 
        {
            "ID": "7890", 
            "Age": "29.1", 
            "Gender": "Male", 
            "DateTime": "7/2/16 4:34"
        }, 
        {
            "ID": "0246", 
            "Age": "85.8", 
            "Gender": "Female", 
            "DateTime": "2/8/12 4:15"
        }, 
        {
            "ID": "0246", 
            "Age": "41.52", 
            "Gender": "Female", 
            "DateTime": "10/11/13 8:41"
        }, 
        {
            "ID": "0246", 
            "Age": "90.52", 
            "Gender": "Female", 
            "DateTime": "10/22/13 8:41"
        }, 
        {
            "ID": "1234", 
            "Age": "74.12", 
            "Gender": "Male", 
            "DateTime": "2/14/14 5:52"
        }, 
        {
            "ID": "3579", 
            "Age": "83.36", 
            "Gender": "Female", 
            "DateTime": "4/15/12 3:33"
        }, 
        {
            "ID": "3579", 
            "Age": "78.36", 
            "Gender": "Female", 
            "DateTime": "4/25/10 3:33"
        }, 
        {
            "ID": "9012", 
            "Age": "26.79", 
            "Gender": "Female", 
            "DateTime": "11/08/12 8:07"
        }, 
        {
            "ID": "4703", 
            "Age": "94.09", 
            "Gender": "Female", 
            "DateTime": "6/10/14 10:10"
        }
    ]},
    "vconcat": [
      {
        "width": 680,
        "layer": [
        {
        "mark": {"type": "line", "tooltip":{"content": "data"}, "color": "black"},
        "encoding": {
          "x": {
            "field": "ID",
            "type": "nominal",
            "sort": null,
            "scale": {"domain":{"selection": "brush", "field": "ID"}, "padding": 5},
            "axis": {"title": "ID"}
          },
          "y": {"field": "Age",
          "type": "quantitative",
          "scale": {"domain": {"selection": "brush", "field": "Age"}, "padding": 5, "zero": false},
          "sort": null
          },
          "detail": {"field": "ID", "type": "nominal"}
        }
    },
    {
        "mark": {"type": "point", "tooltip":{"content": "data"}},
        "encoding": {
          "x": {
            "field": "ID",
            "type": "nominal",
            "sort": null,
            "scale": {"domain": {"selection": "brush", "field": "ID"}, "padding": 5},
            "axis": {"title": "ID"}
          },
          "y": {"field": "Age",
          "type": "quantitative",
          "scale": {"domain": {"selection": "brush", "field": "Age"}, "padding": 5},
          "sort": null
          },
          "color": {"field": "Gender", "type": "nominal"}
        }
    }
    ],
    "resolve": {
        "scale": {
            "color": "independent"
        }
    }
    },
      {
        "width": 680,
        "height": 60,
        "layer":[
        {
        "mark": {"type": "line", "color": "black"},
        "selection": {
          "brush": {
            "type": "interval",
            "encodings": ["x", "y"],
            "on": "[mousedown, window:mouseup] > window:mousemove!",
            "translate": "[mousedown, window:mouseup] > window:mousemove!",
            "zoom": "wheel!",
            "mark": {"fill": "#333", "fillOpacity": 0.125, "stroke": "black"},
            "resolve": "global"
          }
        },
        "encoding": {
          "x": {"field": "ID", 
            "type": "nominal",
            "scale": {"padding": 5},
          "axis": {"title": "ID"},
          "sort": null
        },
          "y": {
            "field": "Age",
            "type": "quantitative",
            "scale": {"padding": 5, "zero": false},
            "axis": {"tickCount": 3, "grid": false, "title": "Age"},
            "sort": null
          },
          "detail": {"field": "ID", "type": "nominal"}
        }
      },
      {
        "mark": {"type": "point"},
        "encoding": {
          "x": {
              "field": "ID", 
            "type": "nominal",
            "scale": {"padding": 5},
          "axis": {"title": "ID"},
          "sort": null
        },
          "y": {
            "field": "Age",
            "type": "quantitative",
            "scale": {"padding": 5},
            "axis": {"tickCount": 3, "grid": false, "title": "Age"},
            "sort": null
          },
          "color": {"field": "Gender", "type": "nominal", "legend": null}
        }
      }
    ],
      "resolve": {
        "scale": {
            "color": "independent"
        }
    }
    }
    ],
    "config": {"axisY": {"minExtent": 30},
  "invalidValues": "filter"}
  }
```

**Comment by nadluru** [comment]
Dear team members,

I was wondering if you had a chance to consider my question above. Is it unclear and/or should I post it via a different mechanism?

Could you kindly provide help and advice?
Thanks!
Nagesh

**Comment by nadluru** [comment]
Dear domoritz and arvind,

I saw that domoritz assigned this thread to arvind. I just wanted to post to this thread to show I am still seeking help in this regard and would truly appreciate any thoughts you are able to share!

Thank you!
Nagesh

**Comment by arvind** [comment]
Dear @nadluru, apologies for never responding to this. As far as I can tell, the x-axis ordering of the top view mirrors the bottom regardless of the direction I brush (left-to-right or right-to-left). So, perhaps we fixed this along the way to Vega 5.17.0, Vega-Lite 4.17.0?
